### DISCUSSION TITLE
Check GPO pins (uart: section), you might need to swap the pins in the config

### DISCUSSION THREAD

**Discussion by opwierde** [gekkekoe/esphome-ecodan-hp#63] [Announcements]
Hi all, might be a silly question but I do not understand how to do the following:
_(check GPO pins (uart: section), you might need to swap the pins in the config)_

There is no data coming from my heat pump yet and I am hoping that this could be the issue. Everything seems to have gone well otherwise (installing, wifi link, esp32s3 status)

**Comment by gekkekoe** [comment]
Sounds like you are using a wrong board definition. Can you post your esphome-ecodan.yaml and specify what hardware you are using and how its wired? If you used the recommended hardware and schematics, it should work out of the box.

**Reply by opwierde** [reply]
![IMG_7099](https://github.com/user-attachments/assets/e29eec5b-b5e4-448a-9473-4de6ad317a6a)
![IMG_7097](https://github.com/user-attachments/assets/f0028599-4421-440c-87ff-ac9dd54a50f7)

As you can see the wiring is a bit confusing due to the plugs I had. 

ESP 32 -> CN105
black  -> red
red  -> white
white  -> yellow
yellow  -> orange

I have compared it to your connections below and tested with ohm meter to be sure Tx and Rx are connected. Power is good as it does come online when plugged into the cn105

grove | cn105
-- | --
pin 1 - black (gnd) | pin 4 - black (gnd)
pin 2 - red (5v) | pin 3 - red (5v)
pin 3 - white (GPIO 2) | pin 2 - white (Tx)
pin 4 - yellow (GPIO 1) | pin 1 - yellow (Rx)



[esphome-ecodan.txt](https://github.com/user-attachments/files/17347133/esphome-ecodan.txt)

**Reply by gekkekoe** [reply]
wires looks okay and if you tested it (you are making yourself hard by using the wires like this), it should be fine. Just to double check, you did check the TX/RX cables from one connector to the other connector right? 

The atom s3 comes online right ? Just not connecting to the heatpump?
When you connect to http://ip_esp what does it say ? Trying to connect messages? you should see the http esphome page.
else post output of
esphome logs --device ip esphome-ecodan.yaml

What you could try is swap white and red (coming from the esp32). So white -> white and red -> yellow. But I don't think that's the issue.

**Reply by opwierde** [reply]
Now I have learned how to switch the wires in a grove connector... next time it will look better.
Swapped the Tx and Rx without result, attached the output on the homepage. A lot of unsuccessful connect messages.
![Screenshot 2024-10-12 at 21 19 49](https://github.com/user-attachments/assets/2f85bde0-390d-40f8-9a94-6ab1f0da1e1d)

**Reply by F1p** [reply]
What are you connecting to? Air-to-Water and what FTC board version 3/4/5/6/7?

**Reply by opwierde** [reply]
Air to water, FTC6. The MEL cloud adapter does work on the CN105 connector. I re-connected it to check.

![IMG_7181](https://github.com/user-attachments/assets/e58f4ed6-6e08-4177-be24-bc42fa1597b3)

**Comment by gekkekoe** [comment]
It tries to send connect sequence and does not get a response (could also be that the connect never gets transmitted over the line). Please check if wire of rx/tx is ok(check at both connector ends)

But it seems okay, what unit and FTC are. you connecting btw? I am running out of ideas why it is not working...

**Reply by opwierde** [reply]
FTC6 board. I made a new cable without the confusing colours. Still no response to a connect attempt... 
![IMG_7183](https://github.com/user-attachments/assets/730c01a3-cefe-4632-aab4-4d1e62644366)

**Reply by F1p** [reply]
Can you show the port you are connecting to on the FTC board please?

**Reply by opwierde** [reply]
Removed the MEL cloud plug and connected the ESP32

![IMG_7181_plug](https://github.com/user-attachments/assets/d3507dbb-65fe-4e31-82ae-52d53720571d)

**Reply by F1p** [reply]
Looks ok to me too, I would check electrical connection of the cable as per the suggestion again


Not to hijack or anything, but you can flash my [MQTT based project](https://github.com/F1p/Mitsubishi-CN105-Protocol-Decode/tree/master/build/esp32.esp32.m5stack-atoms3/v5.2.6/Flash%20and%20Recovery%20Files) and check if you get the same if you wish:

`esptool.exe --chip esp32s3 --port "COM12" --baud 921600  --before default_reset --after hard_reset write_flash -e -z --flash_mode dio --flash_freq 80m --flash_size 4MB 0x0 "ECODAN_Bridge_v5.2.6-h3.bootloader.bin" 0x8000 "ECODAN_Bridge_v5.2.6-h3.partitions.bin" 0xe000 "boot_app0.bin" 0x10000 "ECODAN_Bridge_v5.2.6-h3.bin"
`

You can connect to Telnet and check the connection messages

**Reply by gekkekoe** [reply]
he could try it, but I think RX/TX is broken somehow...

**Reply by F1p** [reply]
> he could try it, but I think RX/TX is broken somehow...

Agree, should make no difference

**Comment by gekkekoe** [comment]
if the melcloud adapter is working then the cn105 is fine. Unit should be supported.
Did you measure/check the RX/TX from both connector ends? (so measure from CN105 connect to the grove connector). 

The cable must be broken or the atom is not okay. Do you have a spare atom ? Else you can map the RX/TX to pin 38/39. But you need to adjust some configs for this (this is assuming that pin 1/2 are broken).

You could try to run the atom outside the unit via usb c cable (and don't connect the cn105, so run it stand alone). And measure the cn105 cable connector RX/TX. it should do 3.3v on TX/RX pins.

**Reply by F1p** [reply]
> You could try to run the atom outside the unit via usb c cable. And measure the grove cable end RX/TX. it should do 3.3v on TX pin.


Warning ⚠️ You must never connect USB at the same time as plugged into the FTC though!

**Reply by gekkekoe** [reply]
notice the run **outside** the unit, stand alone. But edited and clarified it some more

Plug the grove into the esp32, and measure the cn105 connector end
tx and rx should do around 3.3v - 3.4v
Do you see those values? I just tested a spare atom with a correct cable and it should show those values

Also just to check: you did power down the unit, and swapped the melcloud with the esp and turned back the power right?

**Reply by opwierde** [reply]
So, I think I have a broken Atom. The Tx line is below 1 V, the Rx line is around 3.4 V. I will order a replacement and try again later. I appreciate all the effort you guys are putting into my problem!
Yes, I do power down the unit.

**Reply by gekkekoe** [reply]
Double check without wires, does it show the same? You can try to use another pin pair. Never seen a broken atom before btw

**Reply by opwierde** [reply]
Checked without wires as well. Same issue

**Reply by gekkekoe** [reply]
Sorry my bad, I flashed a heishamon firmware (with different pin layout) on one of my test unit atom :) With the correct firmware you should see
TX ~3.4V, but will fluctuate when it's sending connect sequence
RX ~3.4V 

Your readings implies that pins are not correctly configured.

So I think 
1) pin config is wrong
2) your cable is wrong

Try recompile, by cleaning first
```bash
esphome clean ecodan-esphome.yaml
esphome compile ecodan-esphome.yaml
esphome upload --device ip ecodan-esphome.yaml
```

If it still does not work, measure without cable directly on your atom grove port.

**Reply by gekkekoe** [reply]
> Checked without wires as well. Same issue

ah okay then there's something wrong with the TX pin. try clean build install one last time and then order new atom :(
And monitor TX for like 1 mins, each time it tries to connect you should see voltage changing

When you do get a new atom test it standalone and measure voltages if it behaves as expected before plugging it in.

**Reply by opwierde** [reply]
Yes, I will order a new one. I hooked up my scope and the Tx signal looked pretty bad, a lot of ringing so I might have blown the output circuit somehow.
![IMG_7184](https://github.com/user-attachments/assets/13e542c3-9490-478c-9876-5097d736959a)

**Reply by gekkekoe** [reply]
thats quite an achievement, did you wire 12v to it ? :D

**Reply by opwierde** [reply]
Polyester track suit and no anti-static precautions is my favourite method

**Comment by opwierde** [comment]
New Atom and it all works straight out of the box. Thanks for the help!!
